File: src/main/golang/leetcode/editor/cn/[55]Jump Game.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func canJump(nums []int) bool {
	Max := func(x, y int) int {
		if x > y {
			return x
		}
		return y
	}
	n := len(nums)
	var farthest = 0
	for i := 0; i < n-1; i++ {
		farthest = Max(farthest, i+nums[i])
		if farthest <= i {
			return false
		}
	}
	return farthest >= len(nums)-1
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[55]Jump Game.py
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_dis = 0
        for i, num in enumerate(nums):
            if i > max_dis: return False
            max_dis = max(max_dis, i + num)
        return True
        # n = len(nums)
        #
        # def dfs(target: int):
        #     if target >= n - 1:
        #         return True
        #     for i in range(nums[target], 0, -1):
        #         if dfs(target + i):
        #             return True
        #     return False
        #
        # return dfs(0)
# leetcode submit region end(Prohibit modification and deletion)
